lib/Common/CommonDefines.h

```diff
@@ -140,7 +140,11 @@
 #define ENABLE_JS_ETW                               // ETW support
 #else
 #define SYSINFO_IMAGE_BASE_AVAILABLE 0
+#ifndef ENABLE_VALGRIND
 #define ENABLE_CONCURRENT_GC 1
+#else
+#define ENABLE_CONCURRENT_GC 0
```

Comment:
also, if Valgrind supports max 32GB address space, we can change it to  32GB/4096 bytes to be reserved for card table. so you can change AutoSystemInfo to specify max possible address range and still keep concurrent GC enabled